### ISSUE TITLE
RFC: Using Symfony OptionsResolver component

### ISSUE THREAD

**Issue by pjedrzejewski** [thephpleague/omnipay#17]
Hey,

Just very raw idea, you could use [Symfony OptionsResolver component](https://github.com/symfony/optionsresolver) to handle the options definition in gateways.

It's very simple but powerful component. It would allow to define default, required options, check their types, automatically set the defaults if missing and other very cool things.

It also works nicely for gateways which would eventually extend other gateways.

Long story short, it can be used to handle the gateway options in elegant and flexible way.

What do you think?
